### PR TITLE
feat: earn and learn page

### PR DESCRIPTION
**Pull request by carbon-portkey**

### CODE REVIEW

File: packages/mobile-app-did/js/pages/Discover/components/DiscoverTopTab/index.tsx [hunk 1]
```diff
@@ -20,6 +21,10 @@ const DiscoverTab: React.FC = () => {
         name: t('Earn'),
         tabItemDom: <EarnPage />,
       },
+      {
+        name: t('Learn'),
+        tabItemDom: <LearnPage />,
+      },
```
Comment (wade-portkey) [inline packages/mobile-app-did/js/pages/Discover/components/DiscoverTopTab/index.tsx:27]: use data from cms

File: packages/mobile-app-did/js/pages/Discover/components/SubPages/Earn/index.tsx [hunk 2]
```diff
@@ -8,14 +8,17 @@ import { TextL, TextS } from 'components/CommonText';
...
 
 export const EarnPage = () => {
   const { earnList = [] } = useDiscoverData();
   return (
     <View style={styles.container}>
-      {earnList.length > 0 ? earnList.map((item, index) => <EarnItem key={index} {...item} />) : <NoData />}
+      <ScrollView showsVerticalScrollIndicator={false}>
+        {earnList.length > 0 ? earnList.map((item, index) => <EarnItem key={index} {...item} />) : <NoData />}
+        <View style={styles.gap} />
+      </ScrollView>
```
Comment (wade-portkey) [inline packages/mobile-app-did/js/pages/Discover/components/SubPages/Earn/index.tsx:21]: `{earnList.length > 0 ? (
        <ScrollView showsVerticalScrollIndicator={false}>
          {earnList.map((item, index) => (
            <EarnItem key={index} {...item} />
          ))}
          <View style={styles.gap} />
        </ScrollView>
      ) : (
        <NoData />
      )}`
maybe this is better

File: packages/mobile-app-did/js/pages/Discover/components/DiscoverTopTab/index.tsx [hunk 3]
```diff
@@ -20,9 +23,24 @@ const DiscoverTab: React.FC = () => {
...
     ];
   }, [t]);
 
+  const tabList = useMemo(() => {
+    return discoverHeaderTabList
+      .sort((a, b) => Number(a.index) - Number(b.index))
```
Comment (Portkey-David) [inline packages/mobile-app-did/js/pages/Discover/components/DiscoverTopTab/index.tsx:35]: the list data is handled,  just show it

**Review by Portkey-David — APPROVED** [review]
ack

**Pull request merged**